andela/arakari-rc
# #141485311 Marketplace shop layout

#### What does this PR do?

- Adds a shop page that displays all shops available
- Allows users to click on a shop and view all products from that shop
- Allows vendors to set shop Name during registration

 #### Description of Task to be completed?
- Create a page where users are able to view all shops, visit shops and view products from that shop

 #### How should this be manually tested?
- Open arakari-rc deployment
- Click on `Shops`
- Click on any Shop you want

 #### Any background context you want to provide?
There is a bug detected on production: when a vendor registers a shop, an error `
Cannot read property 'language' of undefined` is raised and a 404 page is rendered. This is a bug that I've not been able to figure out. However, after much research I found an issue raised on the same in the main Reaction Commerce repository.
 
The work around I found was to rebuild my heroku app every time this error is raised which bypasses the error and creates the shop as it was supposed to. Not a fun thing to keep doing, thus I would advice anyone trying out the app to avoid creating a shop.

**But what if I want to test it?**
It works perfectly well in development. Just
- `clone the repo`
- run `reaction init`
- `meteor npm install`
- run `reaction`
and create as many shops as you would like :)

Please note that this error occurs only when registering as a vendor/creating a shop thus you can login as a normal user or admin, view shops, buy products and even better, have the vendor experience by logging in to Sindy's quirky shop as:
Email: sindy1234@shop.com
Password: sindy1234 
on the deployed app https://arakari-rc-staging-pr-23.herokuapp.com. 

 #### What are the relevant pivotal tracker stories?
141485311

 #### Screenshots (if appropriate)
Products Home Page

<img width="1440" alt="screen shot 2017-03-23 at 1 39 17 pm" src="https://cloud.githubusercontent.com/assets/16874615/24244031/523d3bdc-0fce-11e7-8f76-95b2203e876f.png">

Shops Home Page
<img width="1439" alt="screen shot 2017-03-23 at 1 39 25 pm" src="https://cloud.githubusercontent.com/assets/16874615/24244075/7bd9e08a-0fce-11e7-9c65-573c47a6c7dc.png">

Sindy's Shop
<img width="1440" alt="screen shot 2017-03-23 at 11 58 22 am" src="https://cloud.githubusercontent.com/assets/16874615/24244108/9a48836e-0fce-11e7-8419-97cd7903bff3.png">

StrapOn's Shop from a Guest view
<img width="1440" alt="screen shot 2017-03-23 at 2 50 16 pm" src="https://cloud.githubusercontent.com/assets/16874615/24246382/2038e50a-0fd8-11e7-9258-8eed87608206.png">

Clarks' Shop
<img width="1440" alt="screen shot 2017-03-23 at 1 39 54 pm" src="https://cloud.githubusercontent.com/assets/16874615/24244145/c77597e6-0fce-11e7-8797-e117566cc303.png">

Teso's Shop
<img width="1439" alt="screen shot 2017-03-23 at 1 39 33 pm" src="https://cloud.githubusercontent.com/assets/16874615/24244180/eb62746c-0fce-11e7-8358-e217ade2a363.png">
